scripts/run_e2e_tests_test.py

```diff
@@ -360,6 +361,108 @@ def test_start_tests_in_debug_mode(self) -> None:
         with self.swap_mock_set_constants_to_default:
             run_e2e_tests.main(args=['--debug_mode'])
 
+    def test_skip_firebase_and_datastore_emulator_when_not_in_emulator_mode(
```

Comment:
Which part of this test is verifying that the firebase + datastore emulator are skipped? Maybe an inline comment might be helpful if it's not obvious.

Reply:
Done